controllers/webhook.js

```diff
@@ -0,0 +1,302 @@
+        await staticman.getSiteConfig().then((siteConfig) => {
+          webhookSecretExpected = siteConfig.get('githubWebhookSecret')
+        })
+      }
+
+      let reqAuthenticated = true
+      if (webhookSecretExpected) {
+        reqAuthenticated = false
+        const webhookSecretSent = req.headers['x-hub-signature']
+        if (!webhookSecretSent) {
+          // This could be worth logging... unless the endpoint gets hammered with spam.
+          errorsRaised.push('No secret found in the webhook request')
+        } else if (_verifyGitHubSignature(webhookSecretExpected, JSON.stringify(req.body), webhookSecretSent)) {
+          reqAuthenticated = true
+        } else {
+          // This could be worth logging... unless the endpoint gets hammered with spam.
+          errorsRaised.push('Unable to verify authenticity of request')
+        }
+      }
+
+      if (reqAuthenticated) {
+        await _handleMergeRequest(req.params, service, req.body, staticman).catch((errors) => {
+          errorsRaised = errors
+        })
+      }
+    }
+  }
+
+  if (errorsRaised.length > 0) {
+    return Promise.reject(errorsRaised)
+  }
+}
+
+const _handleWebhookGitLab = async function (req, service, staticman) {
+  let errorsRaised = []
+
+  const event = req.headers['x-gitlab-event']
+  if (!event) {
+    errorsRaised.push('No event found in the request')
+  } else {
+    if (event === 'Merge Request Hook') {
+      let webhookSecretExpected = null
+      if (staticman) {
+        // Webhook request authentication is NOT supported in v1 of the endpoint.
+        await staticman.getSiteConfig().then((siteConfig) => {
+          webhookSecretExpected = siteConfig.get('gitlabWebhookSecret')
+        })
+      }
+
+      let reqAuthenticated = true
+      if (webhookSecretExpected) {
+        reqAuthenticated = false
+        const webhookSecretSent = req.headers['x-gitlab-token']
+        if (!webhookSecretSent) {
+          // This could be worth logging... unless the endpoint gets hammered with spam.
+          errorsRaised.push('No secret found in the webhook request')
+        } else if (webhookSecretExpected === webhookSecretSent) {
+          /*
+           * Whereas GitHub uses the webhook secret to sign the request body, GitLab does not.
+           * As such, just check that the received secret equals the expected value.
+           */
+          reqAuthenticated = true
+        } else {
+          // This could be worth logging... unless the endpoint gets hammered with spam.
+          errorsRaised.push('Unable to verify authenticity of request')
+        }
+      }
+
+      if (reqAuthenticated) {
+        await _handleMergeRequest(req.params, service, req.body, staticman).catch((errors) => {
+          errorsRaised = errors
+        })
+      }
+    }
+  }
+
+  if (errorsRaised.length > 0) {
+    return Promise.reject(errorsRaised)
+  }
+}
+
+const _calcIsVersion1WhenGitHubAssumed = function (req) {
```

Comment:
```suggestion
const _shouldUseDefaultService = function (req) {
```
nit: I think this would be a slightly more accurate name